src/plugins/dashboard/public/dashboard_app/top_nav/dashboard_top_nav.tsx

```diff
@@ -254,7 +220,45 @@ export function DashboardTopNav({ embedSettings, redirectTo }: DashboardTopNavPr
         ref={dashboardTitleRef}
         tabIndex={-1}
       >{`${getDashboardBreadcrumb()} - ${dashboardTitle}`}</h1>
-      <TopNavMenu {...getNavBarProps()} />
+      <TopNavMenu
+        {...visibilityProps}
+        query={query}
+        screenTitle={title}
+        useDefaultBehaviors={true}
+        indexPatterns={allDataViews}
+        savedQueryId={savedQueryId}
+        showSaveQuery={showSaveQuery}
+        appName={LEGACY_DASHBOARD_APP_ID}
+        visible={viewMode !== ViewMode.PRINT}
+        setMenuMountPoint={embedSettings || fullScreenMode ? undefined : setHeaderActionMenu}
```

Comment:
This is where the actual bug fix is - the logic for this used to simply be:
```typescript
setMenuMountPoint={embedSettings ? undefined : setHeaderActionMenu}
```

By adding `|| fullScreenMode`, we make it so that the portal gets unmounted every time fullscreen mode is entered, then remounted when it is exited :+1: